engine/src/main/java/org/terasology/rendering/opengl/fbms/DynamicFBM.java

```diff
@@ -0,0 +1,163 @@
+import org.lwjgl.opengl.Display;
+import org.lwjgl.opengl.GL11;
+import org.terasology.assets.ResourceUrn;
+import org.terasology.config.Config;
+import org.terasology.config.RenderingConfig;
+import org.terasology.registry.CoreRegistry;
+import org.terasology.rendering.oculusVr.OculusVrHelper;
+import org.terasology.rendering.opengl.AbstractFBM;
+import org.terasology.rendering.opengl.DefaultDynamicFBOs;
+import org.terasology.rendering.opengl.FBO;
+import org.terasology.rendering.opengl.FBOConfig;
+import org.terasology.rendering.opengl.PostProcessor;
+
+/**
+ * TODO: Add javadocs
+ * TODO: Better naming
+ */
+public class DynamicFBM extends AbstractFBM {
+    // I could have named them fullResolution, halfResolution and so on. But halfScale is actually
+    // -both- fullScale's dimensions halved, leading to -a quarter- of its resolution. Following
+    // this logic one32thScale would have to be named one1024thResolution and the otherwise
+    // straightforward connection between variable names and dimensions would have been lost. -- manu3d
+    private FBO.Dimensions fullScale;
+
+    private Config config = CoreRegistry.get(Config.class);
+    private RenderingConfig renderingConfig = config.getRendering();
+    private PostProcessor postProcessor;
+
+    @Override
+    public void initialise() {
+        super.initialise();
+        fullScale = new FBO.Dimensions(Display.getWidth(), Display.getHeight());
+        allocateFBO(DefaultDynamicFBOs.ReadOnlyGBuffer.getFboConfig());
+        allocateFBO(DefaultDynamicFBOs.WriteOnlyGBuffer.getFboConfig());
+        allocateFBO(DefaultDynamicFBOs.Final.getFboConfig());
+    }
+
+
+    /**
+     * Invoked before real-rendering starts
+     * TODO: how about completely removing this, and make Display observable and this FBM as an observer
+     */
+    public void update() {
+        updateFullScale();
+        if (getFBO(DefaultDynamicFBOs.ReadOnlyGBuffer.getResourceUrn()).dimensions().areDifferentFrom(fullScale)) {
+            disposeAllFBOs();
+            createFBOs();
+        }
+    }
+
+    private void disposeAllFBOs() {
+        FBO fbo;
+        for (ResourceUrn urn : fboConfigs.keySet()) {
+            fbo = fboLookup.get(urn);
+            if (fbo != null) {
+                fbo.dispose();
+                fboLookup.remove(urn);
+
+            } else {
+                logger.error("Failed to delete FBO '" + urn + "': it doesn't exist!");
+            }
+        }
+    }
+
+    private void createFBOs() {
+        for (FBOConfig fboConfig : fboConfigs.values()) {
+            generate(fboConfig, fullScale.multiplyBy(fboConfig.getScale()));
+        }
+        notifySubscribers();
+    }
+
+    /**
+     * Returns the content of the color buffer of the FBO "sceneFinal", from GPU memory as a ByteBuffer.
+     * If the FBO "sceneFinal" is unavailable, returns null.
+     *
+     * @return a ByteBuffer or null
+     */
+    public ByteBuffer getSceneFinalRawData() {
```

Comment:
Wouldn't it make sense to move this to the FBO class instead? Perhaps write a TODO about it. We don't need to do it now.